nautobot/core/api/serializers.py

```diff
@@ -29,38 +33,45 @@ def fields(self):
         but `computed_fields` is not specified in the `?include` query parameter, `computed_fields` will be popped
         from the list of fields.
         """
-        fields = super().fields
-        serializer_opt_in_fields = getattr(self.Meta, "opt_in_fields", None)
+        if self.__pruned_fields is None:
+            fields = dict(super().fields)
+            serializer_opt_in_fields = getattr(self.Meta, "opt_in_fields", None)
 
-        if not serializer_opt_in_fields:
-            return fields
+            if not serializer_opt_in_fields:
+                # This serializer has no defined opt_in_fields, so we never need to go further than this
+                self.__pruned_fields = fields
+                return self.__pruned_fields
 
-        if not hasattr(self, "_context"):
-            # We are being called before a request cycle
-            return fields
+            if not hasattr(self, "_context"):
+                # We are being called before a request cycle
+                return fields
+
+            try:
+                request = self.context["request"]
```

Comment:
Did you mean to leave lie 55 in here that sets `request` again? I see that it was there before, but it seems extraneous given this error handling.